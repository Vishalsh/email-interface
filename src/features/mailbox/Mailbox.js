import React, { Component } from 'react';
import PropTypes from 'prop-types';

import classes from './Mailbox.module.scss';

const Mailbox = (props) => {
  const { match } = props;

  return (
    <section className={classes.mailbox}>
      <header className={classes.header}>
        <h1 className={classes.name}>{match.params.mailbox}</h1>
      </header>
    </section>
  );
};

Mailbox.propTypes = {
  match: PropTypes.object.isRequired,
};

export default Mailbox;
