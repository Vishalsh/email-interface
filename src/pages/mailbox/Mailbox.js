import React from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';

import routes from 'constants/routes';
import SettingsCog from 'components/settingsCog/SettingsCog';
import MailboxAside from 'features/mailboxAside/MailboxAside';
import MailboxFeature from 'features/mailbox/Mailbox';

const Mailbox = (props) => {
  const { match } = props;

  return (
    <div className="row">
      <div className="col-sm-3">
        <MailboxAside />
      </div>
      <div className="col-sm-9">
        <Switch>
          <Redirect exact from={routes.MAILBOX} to={routes.INBOX}/>
          <Route path={`${match.url}/:mailbox`} component={MailboxFeature}/>
        </Switch>
      </div>
      <SettingsCog />
    </div>
  );
};

export default Mailbox;
