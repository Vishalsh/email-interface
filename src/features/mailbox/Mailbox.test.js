import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Mailbox from './Mailbox';

describe('Mailbox', () => {
  const props = {
    match: {
      params: {
        mailbox: 'inbox'
      }
    }
  };
  const component = shallow(<Mailbox {...props} />);

  it('should render the component', () => {
    expect(toJson(component)).toMatchSnapshot();
  })
});