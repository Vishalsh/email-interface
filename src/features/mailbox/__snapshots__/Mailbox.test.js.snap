// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mailbox should render the component 1`] = `
<section
  className="mailbox"
>
  <header
    className="header"
  >
    <h1
      className="name"
    >
      inbox
    </h1>
  </header>
</section>
`;
